apps/mobile/src/screens/onboarding/onboarding.tsx

```diff
@@ -0,0 +1,206 @@
+		allAnswers: onboarding.answersByQuestionIndex,
+		currentAnswer:
+			onboarding.answersByQuestionIndex[onboarding.currentQuestionIndex] || "",
+		currentQuestionIndex: onboarding.currentQuestionIndex,
+		dataStatus: onboarding.dataStatus,
+		isLastQuestion:
+			onboarding.currentQuestionIndex ===
+			onboarding.questions.length - PREVIOUS_INDEX_OFFSET,
+		question: onboarding.currentQuestion,
+		totalQuestionsAmount: onboarding.questions.length - PREVIOUS_INDEX_OFFSET,
+	}));
+
+	useEffect(() => {
+		void dispatch(onboardingActions.getAll());
+	}, [dispatch]);
+
+	const { control, errors, handleSubmit, isValid, reset } =
+		useAppForm<OnboardingFormValues>({
+			defaultValues: ONBOARDING_FORM_DEFAULT_VALUES,
+			validationSchema: oneAnswerSelectedValidationSchema,
+		});
+
+	const handleSaveAnswers = useCallback(
+		(payload: OnboardingAnswerRequestBodyDto) => {
+			//TODO: save data to backend
+			return payload;
+		},
+		[],
+	);
+
+	useEffect(() => {
+		reset({ answer: currentAnswer.toString() });
+	}, [currentQuestionIndex, currentAnswer, reset]);
+
+	const handleNextClick = useCallback(
+		(payload: OnboardingFormValues) => {
+			dispatch(
+				onboardingActions.setAnswersByQuestionIndex({
+					answerId: Number(payload.answer),
+					questionIndex: currentQuestionIndex,
+				}),
+			);
+
+			if (isLastQuestion) {
+				handleSaveAnswers({ answerIds: allAnswers });
+
+				return;
+			}
+
+			if (pagerReference.current) {
+				pagerReference.current.incrementPage({ animated: true });
+			}
+
+			void dispatch(onboardingActions.nextQuestion());
+		},
+
+		[
+			isLastQuestion,
+			dispatch,
+			handleSaveAnswers,
+			currentQuestionIndex,
+			allAnswers,
+		],
+	);
+
+	const handlePreviousClick = useCallback(() => {
+		void dispatch(onboardingActions.previousQuestion());
+
+		if (pagerReference.current) {
+			pagerReference.current.decrementPage({ animated: true });
+		}
+	}, [dispatch]);
+
+	const handleFormSubmit = useCallback((): void => {
+		void handleSubmit(handleNextClick)();
+	}, [handleNextClick, handleSubmit]);
+
+	const handleAnalyzePress = useCallback((): void => {
+		void handleSubmit(handleNextClick)();
+		navigation.navigate(RootScreenName.WELCOME);
+	}, [navigation, handleNextClick, handleSubmit]);
+
+	const renderPageComponent = useCallback(() => {
+		return <Content control={control} errors={errors} question={question} />;
+	}, [control, errors, question]);
+
+	return (
+		<LoaderWrapper isLoading={dataStatus === DataStatus.PENDING}>
+			<BackgroundWrapper>
+				<ScreenWrapper>
+					<View
+						style={[
+							globalStyles.flex1,
+							globalStyles.mb16,
+							globalStyles.mh12,
+							globalStyles.mt12,
+							globalStyles.p24,
+							styles.container,
+						]}
+					>
+						{question && (
+							<>
+								<ProgressBar
+									currentItemIndex={currentQuestionIndex}
+									totalItemsAmount={Math.max(
+										totalQuestionsAmount,
+										PREVIOUS_INDEX_OFFSET,
+									)}
```

Comment:
Can we do just like this?

```suggestion
									totalItemsAmount={totalQuestionsAmount}
```